R2API/DifficultyAPI.cs

```diff
@@ -47,6 +48,21 @@ public class DifficultyAPI {
             return AddDifficulty(difficulty, false);
         }
 
+        /// <summary>
+        /// Add a DifficultyDef to the list of available difficulties.
+        /// This must be called before the DifficultyCatalog inits, so before plugin.Start()
+        /// You'll get your new index returned that you can work with for comparing to Run.Instance.selectedDifficulty.
+        /// If this is called after the DifficultyCatalog inits then this will return -1/DifficultyIndex.Invalid and ignore the difficulty
+        /// </summary>
+        /// <param name="difficulty">The difficulty definition to add.</param>
+        /// <param name="difficultyIcon">Sprite to use as the difficulty's icon.</param>
+        /// <returns>DifficultyIndex.Invalid if it fails. Your index otherwise.</returns>
+        public static DifficultyIndex AddDifficulty(DifficultyDef? difficulty, Sprite difficultyIcon) {
```

Comment:
You are accepting a nullable parameter, but your code cannot support a null as input.

Reply:
difficulty is now non-null